third_party/xla_client/xrt_computation_client.cc

```diff
@@ -1405,19 +1414,20 @@ void XrtComputationClient::InitializeDevices(
     // [num_tasks][devices_per_task][mesh_shape_size] coordinates, where the
     // mesh coordinates are usually [x, y, z, c] ('x', 'y' and 'z' being the
     // spatial chip coordinated and 'c' the core number).
-    int64_t base_index = parsed_device.task *
+    int64_t task_index = parsed_device.task *
                              topology_proto->num_tpu_devices_per_task() *
-                             topology_proto->mesh_shape_size() +
-                         parsed_device.id * topology_proto->mesh_shape_size();
+                             topology_proto->mesh_shape_size();
+    int64_t base_index = task_index + parsed_device.id * topology_proto->mesh_shape_size();
     std::vector<int> device_mesh_coords(topology_proto->mesh_shape_size());
     for (int i = 0; i < topology_proto->mesh_shape_size(); ++i) {
       device_mesh_coords[i] =
-          topology_proto->device_coordinates(base_index + i);
+          topology_proto->device_coordinates(base_index - task_index + i);
```

Comment:
Since 
```
base_index = task_index + parsed_device.id * topology_proto->mesh_shape_size()
```

so 
```
base_index - task_index = parsed_device.id * topology_proto->mesh_shape_size()
```

So here it's just 
```
parsed_device.id * topology_proto->mesh_shape_size() + i
```

Then can we simply redefine base_index as 
```
const int64_t base_index = parsed_device.id * topology_proto->mesh_shape_size();
```
and use it here?